inference-engine/tests/functional/plugin/cpu/single_layer_tests/mat_mul.cpp

```diff
@@ -162,43 +205,69 @@ std::vector<fusingSpecificParams> fusingParamsSet3D {
 };
 
 const auto fullyConnectedParams3D = ::testing::Combine(::testing::ValuesIn(IS3D),
-                                                       ::testing::Values(Precision::FP32),
+                                                       ::testing::ValuesIn(netPRCs),
+                                                       ::testing::Values(Precision::UNSPECIFIED),
+                                                       ::testing::Values(Precision::UNSPECIFIED),
+                                                       ::testing::Values(Layout::ANY),
                                                        ::testing::Values(helpers::InputLayerType::CONSTANT),
-                                                       ::testing::ValuesIn(transpose),
-                                                       ::testing::ValuesIn(transpose));
+                                                       ::testing::Values(CommonTestUtils::DEVICE_CPU),
+                                                       ::testing::ValuesIn(additionalConfig));
 
 const auto testParams3D = ::testing::Combine(fullyConnectedParams3D,
                                              ::testing::Values(MatMulNodeType::FullyConnected),
                                              ::testing::ValuesIn(fusingParamsSet3D));
 
-INSTANTIATE_TEST_SUITE_P(smoke_Check_3D, MatMulLayerCPUTest, testParams3D, MatMulLayerCPUTest::getTestCaseName);
+INSTANTIATE_TEST_SUITE_P(smoke_FC_3D, MatMulLayerCPUTest, testParams3D, MatMulLayerCPUTest::getTestCaseName);
 
 }; // namespace fullyConnected
 
 
-/* ============= Gemm ============= */
-namespace gemm {
+/* ============= MatMul ============= */
+namespace matmul {
 
-const std::vector<std::pair<SizeVector, SizeVector>> IS = {
-    {{1, 2, 32, 120}, {120, 5}},
-    {{7, 32, 120}, {3, 7, 120, 50}},
-    {{10, 10, 10}, {10, 10, 10}},
-    {{55, 12}, {12, 55}}
+const std::vector<ShapeRelatedParams> IS = {
+    {{{1, 2, 32, 120}, false}, {{120, 5}, false}},
+    {{{1, 2, 32, 120}, true}, {{120, 5}, false}},
+    {{{1, 2, 32, 120}, false}, {{120, 5}, true}},
+    {{{1, 2, 32, 120}, true}, {{120, 5}, true}},
+
+    {{{7, 32, 120}, false}, {{3, 7, 120, 50}, false}},
+    {{{7, 32, 120}, true}, {{3, 7, 120, 50}, false}},
+    {{{7, 32, 120}, false}, {{3, 7, 120, 50}, true}},
+    {{{7, 32, 120}, true}, {{3, 7, 120, 50}, true}},
+
+    {{{10, 10, 10}, false}, {{10, 10, 10}, false}},
+    {{{10, 10, 10}, true}, {{10, 10, 10}, false}},
+    {{{10, 10, 10}, false}, {{10, 10, 10}, true}},
+    {{{10, 10, 10}, true}, {{10, 10, 10}, true}},
+
+    {{{55, 12}, false}, {{12, 55}, false}},
+    {{{55, 12}, true}, {{12, 55}, false}},
+    {{{55, 12}, false}, {{12, 55}, true}},
+    {{{55, 12}, true}, {{12, 55}, true}},
+};
+
+std::vector<fusingSpecificParams> matmulFusingParams {
+        emptyFusingSpec,
+        fusingRelu,
```

Comment:
As far as I remember all input data are positive in tests, so relu will do nothing. Can we replace with smt else (like elu)?

Reply:
Done